test/geocoders/geocodio.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+
+from geopy import exc
+from geopy.geocoders import Geocodio
+from geopy.point import Point
+from test.geocoders.util import BaseTestGeocoder, env
+
+
+class TestGeocodio(BaseTestGeocoder):
+
+    @classmethod
+    def make_geocoder(cls, **kwargs):
+        api_key = kwargs.pop('api_key') if 'api_key' in kwargs else env['GEOCODIO_KEY']
+        return Geocodio(api_key=api_key, **kwargs)
+
+    async def test_user_agent_custom(self):
+        geocoder = self.make_geocoder(user_agent='my_user_agent/1.0')
+        assert geocoder.headers['User-Agent'] == 'my_user_agent/1.0'
+
+    async def test_error_with_no_api_key(self):
+        with pytest.raises(exc.ConfigurationError):
+            self.make_geocoder(api_key=None)
```

Comment:
You can just create an instance here:
```
    async def test_error_with_no_api_key(self):
        with pytest.raises(exc.ConfigurationError):
            Geocodio(api_key=None)
```

Then it won't be necessary to support retrieving the api_key from kwargs in `make_geocoder`

Reply:
Fixed in 24b45cb265fb267f640f23b6a83baa684c57b5c2